### ISSUE TITLE
OptionValueDurationLearner wraps int32 lifetime counters at INT32_MAX instead of saturating

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#283]
## What's wrong

\`OptionValueDurationLearner.update\` stored lifetime accounting counters (\`option_update_counts\`, \`step_count\`) as \`int32\`, but incremented them with ordinary addition. A valid update at \`INT32_MAX\` therefore wrapped both counters to \`INT32_MIN\`, corrupting the lifetime measurement while the weight update itself was still applied normally.

Reproduced directly before any fix:

\`\`\`text
update_applied=True
option_update_count_before=2147483647
option_update_count_after=-2147483648
step_count_before=2147483647
step_count_after=-2147483648
\`\`\`

## Reachability

\`OptionValueDurationLearner\`, its state/result types, and \`run_option_value_duration_from_arrays\` are imported and listed in \`alberta_framework/core/__init__.py\`. The package root \`alberta_framework/__init__.py\` does not re-export this symbol. The public array runner accepts real observation, option-index, reward, next-observation, and continuation-discount arrays and invokes \`learner.update\` inside \`jax.lax.scan\`; direct calls to the exported learner are also supported.

## Fix

Both counters now saturate at \`INT32_MAX\` via a JAX-compatible \`_saturating_increment\` helper, matching the existing saturating-increment pattern already used elsewhere in this codebase (\`alberta_framework/core/average_reward.py\`'s \`_saturating_int32_increment\`).

PR incoming with a regression test that also asserts the learning weights genuinely change on a saturating update — proving saturation doesn't silently suppress a valid update — plus a mutation-resistance transcript.
